src/software/amazon/ion/impl/bin/IonRawBinaryWriter.java

```diff
@@ -1035,9 +1036,17 @@ public void writeInt(BigInteger value) throws IOException
     public void writeFloat(final double value) throws IOException
     {
         prepareValue();
-        updateLength(9);
-        buffer.writeUInt8(FLOAT_TYPE);
-        buffer.writeUInt64(doubleToRawLongBits(value));
+
+        if (value == ((double) ((float) value))) {
```

Comment:
Minor nit, the code does have a legacy style convention that we should follow (braces on its own line).